packages/wonder-blocks-modal/components/one-pane-dialog/one-pane-dialog.js

```diff
@@ -0,0 +1,152 @@
+
+    /**
+     * The content of the modal's footer. A great place for buttons!
+     *
+     * Content is right-aligned by default. To control alignment yourself,
+     * provide a container element with 100% width.
+     */
+    footer?: React.Node,
+
+    /**
+     * Called when the close button is clicked.
+     *
+     * If you're using `ModalLauncher`, you probably shouldn't use this prop!
+     * Instead, to listen for when the modal closes, add an `onClose` handler
+     * to the `ModalLauncher`.  Doing so will result in a console.warn().
+     */
+    onClose?: () => mixed,
+
+    /**
+     * When set, provides a component that can render content above the top of the modal;
+     * when not set, no additional content is shown above the modal.
+     * This prop is passed down to the ModalDialog.
+     *
+     * NOTE: Devs can customize this content by rendering the component assigned to this prop with custom styles,
+     * such as by wrapping it in a View.
+     */
+    above?: React.Node,
+
+    /**
+     * When set, provides a component that will render content below the bottom of the modal;
+     * when not set, no additional content is shown below the modal.
+     * This prop is passed down to the ModalDialog.
+     *
+     * NOTE: Devs can customize this content by rendering the component assigned to this prop with custom styles,
+     * such as by wrapping it in a View.
+     */
+    below?: React.Node,
+
+    /**
+     * Optional custom styles.
+     */
+    style?: StyleType,
+|};
+
+/**
+ * This is the standard layout for most straightforward modal experiences.
+ *
+ * The ModalHeader is required, but the ModalFooter is optional.
+ * The content of the dialog itself is fully customizable, but the left/right/top/bottom padding is fixed.
+ */
+export default class OnePaneDialog extends React.Component<Props> {
+    renderCloseButton() {
+        const {onClose} = this.props;
+
+        return (
+            <MediaLayout styleSheets={styleSheets}>
+                {({styles}) => (
+                    <CloseButton
+                        light={false}
+                        onClick={onClose}
+                        style={styles.closeButton}
+                    />
+                )}
+            </MediaLayout>
+        );
+    }
+
+    render() {
+        const {
+            onClose,
+            title,
+            subtitle,
+            footer,
+            content,
+            above,
+            below,
+            style,
+        } = this.props;
+
+        return (
+            <MediaLayout styleSheets={styleSheets}>
+                {({styles}) => (
+                    <ModalDialog
+                        style={[styles.wrapper, style]}
+                        above={above}
+                        below={below}
+                    >
+                        <ModalPanel
+                            onClose={onClose}
+                            titleBar={
+                                <Toolbar
```

Comment:
For now I'm going to keep this structure. This will change as part of PR 3/3. Instead of using the `Toolbar` component, we are going to use a refactored version of the `ModalHeader` component. More details will come in PR 3/3.